optimum/onnxruntime/trainer.py

```diff
@@ -850,13 +874,19 @@ def evaluation_loop_ort(
 
             # Update containers on host
             if loss is not None:
+                if self.args.local_rank != -1:
+                    loss = loss.to("cuda")
```

Comment:
Why use the rank and not `self.args.device` ?

Reply:
What if `local_rank != -1` and `args.device="cpu"` ? I think using `args.device` in all cases would solve the issue, what do you think ?

Reply:
Agreed! It would definitely be much better to send the tensors to the device than hard code the device name. Just modified that! Thanks for the suggestion.